uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequest.php

```diff
@@ -347,4 +345,22 @@ public function getBreakdown($leaveRequestID) {
 
     return $breakdown;
   }
+
+  /**
+   * A set of logic to determine whether to skip balance change update
+   * for the leave request being updated or not.
+   *
+   * @param array $params
+   *
+   * @return bool
+   */
+  private function skipBalanceChangeUpdate($params) {
+    $isToilRequest = $params['request_type'] == LeaveRequest::REQUEST_TYPE_TOIL;
+    $datesDidNotChange = $this->datesChanged($params) === false;
+    $leaveRequestDatesNotChanged = !$isToilRequest && $datesDidNotChange;
+    $toilDatesAndToAccrueNotChanged = LeaveRequest::toilToAccrueChanged($params) === false && $datesDidNotChange;
```

Comment:
Do we really need to care about the dates here?

Reply:
Yes, because if we don't, If the dates did change, the previous dates would have been deleted and new dates re-created Here: https://github.com/civicrm/civihr/blob/de275bfda6e3ce241c288436efe818dffbfd32b9/uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php#L806-L806. If the TOIL to accrue did not change, that means the balance change update will be skipped leaving behind new dates with no balance change record.